tests/testthat/test-branching_strategy.R

```diff
@@ -0,0 +1,77 @@
+test_that("infer_feature_from_branch works", {
```

Comment:
as discussed we should add some comments/change test name as to what is being tested

Reply:
added more comments